### PR TITLE
feat: Integrating AudioCenter into bones

### PR DESCRIPTION
**Pull request by RockasMockas**
Moved AudioCenter out of jumpy -> bones.

Reworked to remove the dependency on using a game's settings struct directly + streamlining the interfaces such as having both play_sound and play_music be super easy to use, with extra _advanced and _custom methods for different levels of advanced features/usability on music.

Tested both music/sounds work.

### CODE REVIEW

**Review by MaxCWhitehead — APPROVED** [review]
Adding @zicklag for optional review as I haven't need to look at audio / not too familiar with it, but this all looks great to me.

I will test pulling audio out of Jumpy and using this. Thanks!

**Comment by MaxCWhitehead** [comment]
One question - is there a reason we prefer to have this in `src/renderer/audo.rs` instead of `src/audio.rs`? Maybe want to move into own module?

**Comment by zicklag** [comment]
> One question - is there a reason we prefer to have this in src/renderer/audo.rs instead of src/audio.rs? Maybe want to move into own module?


Oh, yeah, forgot to reply to that. That was from an my original attempts to put all the components that were meant for producing a side-effect outside of the Bones world in `renderer`. That got unintuitive, though, when you think about "rendering" audio or "rendering" controller rumble.

I'm not sure if it's still a good idea to have _some_ module for it, but right now it's just more confusing so we can pull it out of `renderer`.

File: framework_crates/bones_framework/src/render/audio.rs [hunk 1]
```diff
@@ -1,29 +1,364 @@
...
+    },
+    tween,
+    tween::Tween,
+    Volume,
 };
+use tracing::warn;
 
-/// The game plugin for the audio system.
+/// The amount of time to spend fading the music in and out.
+pub const MUSIC_FADE_DURATION: Duration = Duration::from_millis(500);
```
Comment (zicklag) [inline framework_crates/bones_framework/src/render/audio.rs]: Do you think maybe we should make this configurable somehow?

**Review by zicklag — COMMENTED** [review]
Just did a quick skim. Otherwise if it works it's probably fine. :D

File: framework_crates/bones_framework/src/render/audio.rs (hunk 1 above)
Comment (RockasMockas) [reply framework_crates/bones_framework/src/render/audio.rs]: Yeah we can throw it into AudioCenter with a getter/setter + default.

**Pull request merged**